### ISSUE TITLE
Every async'ly executing method must return a Future-like object

### ISSUE THREAD

**Issue by PawelLipski** [VirtusLab/git-machete-intellij-plugin#620]
This could prevent errors like https://github.com/VirtusLab/git-machete-intellij-plugin/pull/619/files (and possibly more serious ones) from ever happening (since the programmer would instantly see which method executes some stuff in the background).

Callback hell as e.g. around `com.virtuslab.gitmachete.frontend.ui.impl.table.EnhancedGraphTable#refreshModel` is becoming hard to maintain.

Instead of directly invoking `com.intellij.openapi.progress.Task#queue` we should use some abstraction that would return a handle to the async computation (most likely, some kind of `Future`/`Promise` - TBD what's currently the best choice for us, possibly just Vavr Future since we use Vavr already + it has a familiar Scala-like API).

**Comment by PawelLipski** [comment]
Let's first try rewriting a single Task.Backgroundable to Futures as a PoC to make sure it really makes sense first!

I still take into account the possible outcome of _sorry, we need to leave stuff as is_ ☹️

**Comment by azdrojowa123** [comment]
I tried to rewrite single Task.Backgroundable and I ran into a few problems that are hard for me to jump to currently e.g. somehow override onSucess/onThrowable from single Task.Backgroundable (means that default onSuccess should be in single Task.Backgroundable and from time to time override it in specific situation), custom annotations on callbacks e.g. @UIEffect, wrap queue() method to something different because queue() return void (only wrapping is not a big deal but propagating exceptions from run() method into this wrapped one will be probably a problem).
